packages/auth/test/helpers/integration/helpers.ts

```diff
@@ -116,3 +117,37 @@ export const email = 'totpuser-donotdelete@test.com';
 export const fakePassword = 'password';
 //1000000 is always incorrect since it has 7 digits and we expect 6.
 export const incorrectTotpCode = '1000000';
+
+/**
+ * Generates a valid password for the project or tenant password policy in the Auth instance.
+ * @param auth The {@link Auth} instance.
+ * @returns A valid password according to the password policy.
+ */
+export async function generateValidPassword(auth: Auth): Promise<string> {
+  if (getEmulatorUrl()) {
+    return 'password';
+  }
+
+  // Fetch the policy using the Auth instance if one is not cached.
+  const authInternal = _castAuth(auth);
+  if (!authInternal._getPasswordPolicyInternal()) {
+    await authInternal._updatePasswordPolicy();
+  }
+
+  const passwordPolicy = authInternal._getPasswordPolicyInternal()!;
+  const options = passwordPolicy.customStrengthOptions;
+
+  // Create a string that satisfies all possible options.
```

Comment:
nit: clarify what the options are?

"// Create a string that satisfies all possible options(uppercase, lowercase, number, special char)."

Reply:
Done.